app/models/user.rb

```diff
@@ -90,6 +99,22 @@ def missing_required_fields?
 
   private
 
+  def if_driver_remove_unassigned(role)
+    if self.is_a_driver?
+      self.remove_role(:unassigned_driver)
+    end
+  end
+
+  def make_unassigned(role)
+    unless self.is_a_driver?
```

Comment:
this seems broad - are you passed in the role that was removed? should this only do unassigned_driver if the role that was removed had type :driver?